src/mpid/ch4/src/ch4_progress.h

```diff
@@ -168,6 +175,11 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_progress_test(MPID_Progress_state * state, in
 MPL_STATIC_INLINE_PREFIX void MPIDI_progress_state_init(MPID_Progress_state * state)
 {
     state->flag = MPIDI_PROGRESS_ALL;
+    /* For lockless, no VCI lock is needed during NM progress */
+    if (MPIDI_CH4_MT_MODEL == MPIDI_CH4_MT_LOCKLESS) {
+        state->flag &= ~MPIDI_PROGRESS_NM_VCI_LOCK;
+    }
```

Comment:
Both statements effectively achieves the same thing. `state->flag &= ~MPIDI_PROGRESS_NM_VCI_LOCK;` should reset just a single bit from `MPIDI_PROGRESS_NM_VCI_LOCK`, while keeping other bits unchanged. 

Yet, I'm fine switching to this alternative version, if the simplification is a good enough motivations (i.e., set MPIDI_PROGRESS_NM_LOCKLESS only for lockless instead of setting MPIDI_PROGRESS_NM_VCI_LOCK for non-lockless cases). Any thoughts on whether keep it as is or change?

Reply:
Yeah, you are right that they achieve the same thing. I much prefer a simple `|`. Orthogonal bits is easier to read IMO.